tests/lib/screenshot-testing/run-tests.js

```diff
@@ -43,8 +43,8 @@ async function main() {
 
     mocha = new Mocha({
         ui: 'bdd',
-        reporter: config.reporter,
         bail: false,
+        reporter: 'node_modules/@testomatio/reporter/lib/adapter/mocha',
```

Comment:
I heard back from testomatio, here's updates to your feedback

Mocha update is required because in mocha 8 they introduced constants for passed/failed tests. We rely on these constants and probably this is the only issue here. Even, we may probably support old mocha versions it's a dead end path as software should be constantly updated.

1. Our reporter should do some output in a spec-like manner, so no multi-report should be needed (see screenshot)
2. Yes, reporter won't work without API key. I think we may remove that requirement, as mocha reporter still can print output data without a key. We can update it!
3. Yes, you can pass API key to job action as an environment variable and that will enable reporting
![image (4)](https://github.com/matomo-org/matomo/assets/141092099/5d8cdf6e-600d-4c5b-8f4a-e17e1dd64f4e)

Reply:
For #2, that should be done on testomatio side. 

For #3, @matomo-org/core-team do you know how to pass the testomatio environment variable that Thomas created into the Github actions job? I think we use custom actions to run tests, can a custom action load TESTOMATIO secret and set it as environment variable?

Reply:
If you take a look at some of the existing github actions they [retrieve secrets from environment variables](https://github.com/matomo-org/matomo/blob/a89e26dd7e05b56ce453796e86f81637b7a9808a/.github/workflows/matomo-tests.yml#L61).  `${{ secrets.TESTOMATIO }}` should get the API key in the action and then you should be able to pass it to mocha using a .env file and finally access it in JavaScript with ```process.env.TESTOMATIO```

So in the github action do something like:
```
run: |
    touch .env
    echo TESTOMATIO_API_KEY=${{ secrets.TESTOMATIO }} >> .env
```
then in `run-tests.js` you should be able to do:
```
let api_key = process.env.TESTOMATIO;
```

Reply:
Regarding the output:

Yeah, the adapter does some output, but, from [what I can see](https://github.com/testomatio/reporter/blob/master/lib/adapter/mocha.js), it is subpar to the regular spec output (missing indentation, missing describe blocks, no test runtimes). However, I haven't run that one locally, so please take that with a big grain of salt.

Also two days ago there was a [change](https://github.com/testomatio/reporter/commit/2f92d655f2ff8a746327151208a2d919c9b953ea) to the reporter, so it should now always output something, even if the API key is missing. Without that change we will probably still need a solution for non-CI environments, e.g. a fallback to the regular spec reporter if no API key is found, but multi-reporter should be unnecessary and only create a complete output mess.

Reply:
> If you take a look at some of the existing github actions they [retrieve secrets from environment variables](https://github.com/matomo-org/matomo/blob/a89e26dd7e05b56ce453796e86f81637b7a9808a/.github/workflows/matomo-tests.yml#L61). `${{ secrets.TESTOMATIO }}` should get the API key in the action and then you should be able to pass it to mocha using a .env file and finally access it in JavaScript with `process.env.TESTOMATIO`
> 
> So in the github action do something like:
> 
> ```
> run: |
>     touch .env
>     echo TESTOMATIO_API_KEY=${{ secrets.TESTOMATIO }} >> .env
> ```
> 
> then in `run-tests.js` you should be able to do:
> 
> ```
> let api_key = process.env.TESTOMATIO;
> ```

Added the environment variable to matomo-tests.yml , tests ran and failed (expected I think), but I don't see the reporting in testomat. I am going to follow up with testomat for this one.

Reply:
> Added the environment variable to matomo-tests.yml

That file wasn't changed in this PR, so not sure what you tried to do

Reply:
Hi @sgiehl I was able to clean up my local branches and made sure my commits made it in this time :) Thanks for clearing that up. Now that the change is in, somehow I managed to not run all the required tests. I think I did not update the actions correct in matomo-tests.xml. Going to review with the testomat team.